placeholder for 2fa guide (#506)
CHTC/chtc-website-source: uw-research-computing/configure-ssh.md

<iframe width="560" height="315" src="https://www.youtube.com/embed/J-wxsrQ3v04" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## Re-Using SSH Connections

To reduce the number of times it is necessary to enter your credentials, it’s possible 
to customize your ssh configuration in a way that allows you to “reuse” a connection 
for logging in again or moving files.  This configuration is totally optional, and 
most useful if you will connect to 
the same server multiple times in a short window, for example, when uploading or 
downloading files. 

**WARNING:** This guide describes how to configure your local machine to not require 
reentering your NetID password or Duo authentication each time you login. 
This should **ONLY** be used on secure devices that you manage - **it should 
not be used on any shared laptop, desktop, or research group resource. Users 
found violating this policy risk having their CHTC account permanently deactivated.** 

The instructions below are meant for users who can use a terminal (Mac, Linux): 

1. Open a terminal window.

> 1. Create (or edit) your personal ssh configuration file at `~/.ssh/config` to use 

Author: this won't fix it, if they aren't.

what’s called "[ControlMaster](https://www.anchor.com.au/blog/2010/02/ssh-controlmaster-the-good-the-bad-the-ugly/)" (third party blog link). 
Executing the code below will add the right information to the `config` file; you 
can also copy and paste the content in quotes into your config file directly. 
	```	
	# Let's create (or add to) our SSH client
	# configuration file. 
	echo "
	Host *.chtc.wisc.edu
	  # Turn ControlMaster on
	  ControlMaster auto
	  # ControlMaster connection will persist
	  # for 2 hours of idleness, after which
	  # it will disconnect
	  ControlPersist 2h
	  # Where to store files that represent
	  # the ControlMaster persistent connections
	  ControlPath ~/.ssh/connections/%r@%h:%p" >> ~/.ssh/config
	```
	{:.term}

1. You also create a directory that will be used to track connections. In 
the same .ssh directory, make a folder called `connections` by typing: 
```
$ mkdir -p ~/.ssh/connections

> ```{: .term}

Reviewer: Should there be an empy space between line 68/69? Unsure if this is meant to be in-text code or a code chunk (either way is fine)

Once you login to a CHTC server, this is where the system will store information 
about your previous connection information so that you do not have to reenter your 

> password or Duo Authenticate. 

Reviewer: ```suggestion
password or Duo authenticate. 
```

> 1. Now, log into a CHTC node as normal. The first time you log in, you will need to use 

Reviewer: ```suggestion
4. Now, log into your CHTC submit server or login node as normal. The first time you log in, you will need to use 
```

> two factor authentication, but subsequent logins **to that node** will not require 

Reviewer: ```suggestion
two-factor authentication, but subsequent logins **to that machine** will not require 
```

authentication as long as they occur within the time value used in 
the `ControlPersist` configuration option (so in this example, 2 hours). 

> For Windows users who use Putty to log in, you need to go to 

Reviewer: ```suggestion
For Windows users who use PuTTY to log in, you need to go to 
```

the `Connection -> SSH` section in the “Category” menu on the left side, 
and then check the “Share SSH Connection if possible” box. If you don’t 

> see this option, try downloading a newer version of Putty.  

Reviewer: ```suggestion
see this option, try downloading a newer version of PuTTY.  
```
